common/src/main/java/common/BallerinaModel.java

```diff
@@ -871,6 +871,14 @@ public String toString() {
                         "\n";
             }
         }
+
+        /**
+         * This is used to represent a statement that is not fully constructed yet.
+         *
+         * @param statement the statement that is partially constructed
+         */
+        record PartialStatement(Statement statement) implements Statement {
```

Comment:
It is not entirely clear to me why we need this. From what I can see, we seem to use this as a marker for some DoStatements. Is that correct?